test/monitoringlocationlinter/analyzer.go

```diff
@@ -0,0 +1,83 @@
+package monitoringlocationlinter
+
+import (
+	"go/ast"
+	"strings"
+
+	"golang.org/x/tools/go/analysis"
+)
+
+const (
+	prometheusImportPath = `"github.com/prometheus/client_golang/prometheus"`
+)
+
+// NewAnalyzer returns an Analyzer.
+func NewAnalyzer() *analysis.Analyzer {
+	return &analysis.Analyzer{
+		Name: "monitoringlocationlinter",
+		Doc:  "Ensures that in Kubernetes operators projects, monitoring-related practices are implemented within pkg/monitoring directory.",
+		Run:  run,
+	}
+}
+
+// run is the main assertion function.
+func run(pass *analysis.Pass) (interface{}, error) {
+	for _, file := range pass.Files {
+		if !isInMonitoringDir(pass.Fset.File(file.Pos()).Name()) {
+			lintFile(pass, file)
+		}
+	}
+
+	return nil, nil
+}
+
+// lintFile reports if there are metrics registrations in the file.
+func lintFile(pass *analysis.Pass, file *ast.File) {
+	prometheusLocalName := getPrometheusLocalName(file)
+	if prometheusLocalName == "" { // no prometheus import => no use in prometheus in this file; nothing to do here
+		return
+	}
+
+	ast.Inspect(file, func(node ast.Node) bool {
+		call, ok := node.(*ast.CallExpr)
+		if ok && isRegisterMetricsCall(call, prometheusLocalName) {
+			pass.Reportf(node.Pos(), "monitoring-location-linter: Prometheus metrics should be registered only within pkg/monitoring directory.")
+			return true
+		}
+
+		return true
+	})
+}
+
+// isInMonitoringDir returns true if the file is located under pkg/monitoring directory.
+func isInMonitoringDir(filePath string) bool {
+	return strings.Contains(filePath, "pkg/monitoring")
+}
```

Comment:
@machadovilaca other directories to exclude?

Reply:
actually `prometheus.Register` and `prometheus.MustRegister` should not be used in any directory

we should then you can add a new rule, where `operatormetrics.RegisterMetrics`, `operatorrules.RegisterAlerts` and `operatorrules.RegisterRecordingRules` are those yes, only used in `pkg/monitoring`

these functions are the ones we now use to managed observability resources

Reply:
+1, will update.

Reply:
Updated.